onyo/utils.py

```diff
@@ -190,72 +190,97 @@ def check_float(value):
         return False
 
 
-def edit_file(file, onyo_root, onyo_new=False):
+def create_onyo_temp_file(file, onyo_root):
     """
-    Check if the selected editor exists. Check if a file exists, create a copy
-    of it in .onyo/temp/ to change with the editor, and check the temp file
-    after editing for correct yaml syntax and validity specified by the onyo
-    validation file.
-    If the asset is not valid, open a dialog to enable to correct the asset, or
-    remove the changes.
-
-    Returns True on success.
+    Decide for an existing asset file, if a temporary file needs to be created,
+    or if one already exists and should be used.
+    If no temporary file exists, create a copy of the given asset.
+    If one exists, ask the user if it should be further edited, or if the
+    changes should be discarded, in which case the temporary file gets
+    overwritten by a new copy of the input asset.
+
+    Returns on success the Path.object of the temporary asset file.
     """
-    editor = get_editor(onyo_root)
-    # verify that the editor exists
-    if not shutil.which(editor):
-        logger.error(f"The editor '{editor}' was not found. Exiting.")
-        sys.exit(1)
-
-    # verify existence of file to edit
-    file = Path(file)
-    if not file.is_file():
-        logger.error(f"{file} does not exist.")
-        sys.exit(1)
-
-    # create and edit a temporary file, and if that is valid replace original
     temp_file = Path(get_git_root(onyo_root), ".onyo/temp/", file.name)
+
+    # if no temporary file exists, just create a copy of the input file
     if not temp_file.is_file():
         shutil.copyfile(file, temp_file)
-    # When temp-file exists, ask if to use it
-    elif temp_file.is_file():
+
+    # Otherwise ask if it should be used, or overwrite it with a new copy of the
+    # asset
+    else:
         while True:
-            edit_temp = str(input(f"Temporary changes for {file} exist. Continue editing? (y/N) "))
-            if edit_temp in ['y', 'Y', 'yes']:
+            edit = input(f"Temporary changes for {file} exist. Continue editing? (y/N) ")
+            if edit in ['y', 'Y', 'yes']:
                 break
-            elif edit_temp == 'n':
+            elif edit == 'n':
                 shutil.copyfile(file, temp_file)
                 break
+    # return the temporary file
+    return temp_file
+
+
+def validate_yaml_syntax(file):
```

Comment:
`valid_yaml()` or `is_valid_yaml()` are shorter and will read more naturally in an `if` statement.